### ISSUE TITLE
TextArea clear() changes also component height

### ISSUE THREAD

**Issue by lubomirmatus** [vaadin/framework#11593]
Hi

I'm using vaadin framework 8.8.0, on win7 x64, tomee 7.

I have found problem with `TextArea` component. When height is set to 100% or componet is added to `VerticalLayout` using `addComponentAndExpand(...)` method. Then when you call `clear()` method, the component height is changed. I have used it in dialog window and after `clear()` the component height raised and bottom components has disapeared out of dialog window.
I have to set fixed height in pixels to text-area. Then everything works fine.

Example:
```
VerticalLayout v = new VerticalLayout();

//... add some components to layout above text area ...

TextArea ta = new TextArea();
ta.addStyleName(ValoTheme.TEXT_AREA_TINY);
// v1: set height 100% to fill the rest of window
ta.setHeight(100, Unit.PERCENTAGE);
v.addComponent(ta);
// v2: set component to expand in window
v.addComponentsAndExpand(ta);

//... add some footer components to layout after text area ...
```

Then `ta.clear();` in event handler will change the `ta` height and break the layout.

**Comment by TatuLund** [comment]
This should be retested as it may have been potentially fixed by https://github.com/vaadin/framework/pull/11912

**Comment by tarekoraby** [comment]
The issue is still present in version 8.10.2. The problem occurs if the expanding component is added either vertically or horizontally, and the issue also occurs for both a TextField as well as a TextArea (didn't test everything, but a Button expands without a problem). 

However, a workaround is possible: if one adds the TextArea or TextField to another layout and then add this layout to the parent layout, then the component seems expand normally. For example, for a TextArea one could do like so:

```
VerticalLayout layout = new VerticalLayout();

// ... Add some components to the layout

VerticalLayout embededLayout = new VerticalLayout();
embededLayout.setSpacing(false);
embededLayout.setMargin(false);
TextArea textArea = new TextArea("TextArea");
textArea.setValue("Some text");

embededLayout.addComponentsAndExpand(textArea);
layout.addComponentsAndExpand(embededLayout);

Button button = new Button("Change Text");
button.addClickListener(e -> {
	textArea.setValue("Some totaly different text");
});
layout.addComponent(button);

setContent(layout);
```

**Comment by tarekoraby** [comment]
Issue is now fixed in version 8.10.3
